lib/class-wp-rest-url-details-controller.php

```diff
@@ -118,7 +118,10 @@ public function parse_url_details( $request ) {
 
 		$data = $this->add_additional_fields_to_object(
 			array(
-				'title' => $this->get_title( $remote_url_response ),
+				'title'       => $this->get_title( $remote_url_response, $url ),
+				'icon'        => $this->get_icon( $remote_url_response, $url ),
+				'description' => $this->get_description( $remote_url_response, $url ),
+				'image'       => $this->get_image( $remote_url_response, $url ),
```

Comment:
The work being done here is only in the `<head>`.  To make the string searches faster and more efficient, I'd suggest splitting the HTML to process only the  `<head>`.

How?

Add a new private method to search for the closing `</head>` tag or if missing, the opening `<body>` tag.  Then return the string up to that tag. Then pass this partial HTML to each of these metadata helper methods.

Reply:
Agreed. I've added a basic implementation that gets everything before the opening `<body>` tag. That should be good enough for our purposes.

Reply:
...although I guess if there wasn't a `<body>` tag that could be an issue.